source/Calamari.CloudAccounts/AwsAuthenticationProvider.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using System.Linq;
+using System.Security.Authentication;
+using System.Text;
+using System.Threading.Tasks;
+using Amazon;
+using Amazon.ECR;
+using Amazon.ECR.Model;
+using Amazon.Runtime;
+using Amazon.SecurityToken;
+using Amazon.SecurityToken.Model;
+using Calamari.Common.Plumbing.Variables;
+
+namespace Calamari.CloudAccounts
+{
+    public static class AwsAuthenticationProvider
+    {
+        const string DefaultSessionName = "OctopusAwsAuthentication";
+           public static async Task<(string Username, string Password, string RegistryUri)> GetEcrOidcCredentials(IVariables variables)
+        {
+            try
+            {
+                var jwt = variables.Get(AuthenticationVariables.Jwt);
+                var roleArn = variables.Get(AuthenticationVariables.Aws.RoleArn);
+                var region = variables.Get(AuthenticationVariables.Aws.Region);
+                var sessionDuration = variables.Get(AuthenticationVariables.Aws.SessionDuration);
+                
+                var client = new AmazonSecurityTokenServiceClient(new AnonymousAWSCredentials());
```

Comment:
Is there an option to pass a HttpHandler/client in here? Tentacle has some proxies configurations that I don't think would be picked up here. This [used to be an issue for ARM steps](https://github.com/OctopusDeploy/Calamari/pull/1150/files).